lib/cisco_node_utils/dns_domain.rb

```diff
@@ -59,8 +59,24 @@ def create
         config_set('dnsclient', 'domain_list',
                    state: '', name: @name)
       else
-        config_set('dnsclient', 'domain_list_vrf',
-                   state: '', name: @name, vrf: @vrf)
+        # On some platforms attempts to create a new domain-list results
+        # in the error. 'ERROR: Deletion of VRF test in progresswait
+        # for it to complete'.  We handle this by trying up to 500 times
+        # before giving up.
+        tries = 500
```

Comment:
I'd definitely prefer a sleep of 1 s (or even 0.5 s if that's straightforward to do). If the slowest you've seen is 3 seconds, I'd rather retry 10 times with a 0.5s delay (or 5 times with a 1s delay) than 120 times in the same time period.